src/Aspire.Hosting/ApplicationModel/ResourceLoggerService.cs

```diff
@@ -244,9 +249,28 @@ void Log(LogLine log)
 
             try
             {
-                await foreach (var entry in channel.GetBatchesAsync(cancellationToken: cancellationToken).ConfigureAwait(false))
+                var lineCount = 0;
+                while (await channel.Reader.WaitToReadAsync(cancellationToken).ConfigureAwait(false))
                 {
-                    yield return entry;
+                    var batch = new List<LogLine>();
+                    while (channel.Reader.TryRead(out var log))
+                    {
+                        if (lineCount > backlogSnapshot.Length)
```

Comment:
Do IIUC that each logged line, even if it showing up in the snapshot or the channel here, would have the same `log.LineNumber`? And can we expect these to be in order? If so, maybe we 
can use that fact to decide when we are "past", like if the `log.LineNumber > backlogSnapshot[^1].LineNumber`, and thus avoid `backlogSnapshot.Contains`?

I'm super new to the code, jfyi! 😬

Reply:
Yeah I attempted an approach like that first but had issues with it while this worked. Admittedly I was still tracking down the root cause at that point and then @eerhardt made the breakthrough and this was the logic we landed on.